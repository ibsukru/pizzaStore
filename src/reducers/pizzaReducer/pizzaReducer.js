const pizzaReducer = (state = null, action) => {
  switch (action.type) {
    case 'SIZE_SELECT':
      return action.payload
    case 'TOPPING_SELECT':
      const { pizza, topping } = action.payload
      return Object.assign({ ...pizza }, {
        toppings: pizza.toppings.map(current => {
          if (current.topping.name === topping.name) {
            return Object.assign({ ...current }, { defaultSelected: !current.defaultSelected })
          }

          return current
        })
      })
    case 'CLEAR_CLEAR':
      return null
    default:
      return state
  }
}

export default pizzaReducer
