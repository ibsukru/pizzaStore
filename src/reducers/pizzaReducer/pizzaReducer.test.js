/* eslint-env jest */
import pizzaReducer from './pizzaReducer'

[
  {},
  { state: null, action: null },
  {
    state: {},
    action: {
      type: 'UNKNOWN'
    }
  },
  {
    action: {
      type: 'CLEAR_ORDER'
    }
  },
  {
    state: {},
    action: {
      type: 'TOPPING_SELECT',
      payload: { name: 'large' }
    }
  },
  {
    state: null,
    action: {
      type: 'TOPPING_SELECT',
      payload: {
        pizza: null,
        topping: {
          name: 'pepperoni'
        }
      }
    }
  },
  {
    state: {},
    action: {
      type: 'TOPPING_SELECT',
      payload: {
        pizza: {
          toppings: [
            {
              defaultSelected: false,
              topping: {
                name: 'pepperoni',
                price: 0.4
              }
            }
          ]
        },
        topping: {
          name: 'pepperoni'
        }
      }
    }
  }
].forEach(({ state, action }) =>
  test(`Reduce action for state ${JSON.stringify(state)} and action ${JSON.stringify(action)}`, () => {
    expect(pizzaReducer(state, action)).toMatchSnapshot()
  })
)
