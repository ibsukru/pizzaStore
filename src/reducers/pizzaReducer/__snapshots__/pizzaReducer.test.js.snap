// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reduce action for state {} and action {"type":"TOPPING_SELECT","payload":{"name":"large"}} 1`] = `Object {}`;

exports[`Reduce action for state {} and action {"type":"TOPPING_SELECT","payload":{"pizza":{"toppings":[{"defaultSelected":false,"topping":{"name":"pepperoni","price":0.4}}]},"topping":{"name":"pepperoni"}}} 1`] = `
Object {
  "toppings": Array [
    Object {
      "defaultSelected": true,
      "topping": Object {
        "name": "pepperoni",
        "price": 0.4,
      },
    },
  ],
}
`;

exports[`Reduce action for state {} and action {"type":"UNKNOWN"} 1`] = `Object {}`;

exports[`Reduce action for state null and action {"type":"TOPPING_SELECT","payload":{"pizza":null,"topping":{"name":"pepperoni"}}} 1`] = `null`;

exports[`Reduce action for state null and action null 1`] = `null`;

exports[`Reduce action for state undefined and action {"type":"CLEAR_ORDER"} 1`] = `null`;

exports[`Reduce action for state undefined and action undefined 1`] = `null`;
